### PR TITLE
Nativescript 7.x support

### LINKED ISSUE #114: Warning: topmost() is deprecated. Use Frame.topmost() instead.
### Version numbers:
- **CLI:** 6.5.0
- **Cross-platform modules:** 6.5.1
- **Runtime(s):**
```json
"tns-android": {
      "version": "6.0.0"
    },
    "tns-ios": {
      "version": "6.0.1"
    }
```
- **Plugin(s):**
```json
  "dependencies": {
    "nativescript-vue": "^2.5.1",
    "@nativescript/theme": "^2.2.1",
    "nativescript-oauth2": "^2.4.0",
    "tns-core-modules": "^6.0.0",
    "electron": "^8.2.1",
    "email-validator": "^2.0.4",
    "vue-class-component": "^7.2.3",
    "vue-property-decorator": "^8.4.1"
  },
  "devDependencies": {
    "@babel/core": "^7.0.0",
    "@babel/preset-env": "^7.0.0",
    "@types/node": "^12.12.16",
    "@typescript-eslint/eslint-plugin": "^2.26.0",
    "@typescript-eslint/parser": "^2.26.0",
    "@vue/cli-plugin-eslint": "^4.2.3",
    "@vue/eslint-config-airbnb": "^5.0.2",
    "@vue/eslint-config-standard": "^5.1.0",
    "babel-loader": "^8.0.2",
    "eslint": "^6.8.0",
    "eslint-config-airbnb-base": "^14.1.0",
    "eslint-config-standard": "^14.1.1",
    "eslint-plugin-import": "^2.20.2",
    "eslint-plugin-node": "^11.1.0",
    "eslint-plugin-promise": "^4.2.1",
    "eslint-plugin-standard": "^4.0.1",
    "eslint-plugin-vue": "^7.0.0-alpha.0",
    "lint-staged": "^9.5.0",
    "nativescript-dev-webpack": "^1.0.0",
    "nativescript-vue-template-compiler": "^2.5.0",
    "nativescript-worker-loader": "~0.9.0",
    "node-sass": "^4.9.2",
    "prettier": "^2.0.2",
    "tns-platform-declarations": "^6.0.0",
    "tslint": "^6.1.1",
    "typescript": "^3.8.3",
    "vue": "^2.5.22",
    "vue-loader": "^15.4.0"
  }
```

### Steps to reproduce it.
Follow any tutorial with ^6.5.0 versions of {N}.
Run:
```javascript
client = new TnsOAuthClient(providerType, true)
client.loginWithCompletion((tokenResult: ITnsOAuthTokenResult, error) => {
    if (error) {
        console.log(error)
    } else {
        console.log(tokenResult)
    }
})
```
You get in console:
```topmost() is deprecated. Use Frame.topmost() instead.```

### Solution
In both ```loginWithCompletion``` and ```logoutWithCompletion``` in [oauth.ts](https://github.com/alexziskind1/nativescript-oauth2/blob/master/src/oauth.ts#L67), change:
```javascript
frameModule.topmost()
```
For:
```javascript
frameModule.Frame.topmost()
```

### LINKED ISSUE #124: Support for next nativescript release
Hi, 

Recently the nativescript team has started to update basically all their packages. That is particularly important if you are using Angular >= 9 because it let's you benefit of various new features, one of them the new ng compiler. 

Unfortunately I was not able to use the current version of this plugin, but with some little changes it can be compiled with typescript@3.9 and so become compatible with new nativescript build toolchain (which is still under development, stable release should be published in the next days, from what I read).

I've forked the project and pushed the changes, I don't know if the author of this plugin has some time to review them and in case merge with the actual version, just for the people who need a working version and can't wait for the official ns packages to be released. 

Unfortunately I've tried to launch tests but there are changes also on that side, so before to continue and fix them too, let's check what @alexziskind1  thinks about it :-) 

Cheers

### PR DESCRIPTION
**Pull request by funder7**
## What is the current behavior?
The plugin is not usable inside a NativeScript project using Angular version >= 9

## What is the new behavior?
Dependencies have been updated in order to make the project compatible with typescript@3.9. Apparently this ts version is required to allow the import inside a Nativescript-Angular project which is using the new Ivy compiler since angular@9

## Important notes ##
- This code has been included, built, and run into an existing NS project. At runtime it showed some warnings, but all errors were solved. 
- Test configuration must be updated in order to make tests runnable again, don't mark this code as "stable" before enabling tests again. 
- This change: `qsObj["code"]` --> `qsObj.code.toString()` has been tested on Android and works, by the way some tests would be a nice to have.

Fixes #124 .
Fixes #114 .

### CODE REVIEW

**Comment by alexziskind1** [comment]
@funder7 thanks for the PR. is this backwards compatible with ns6?

**Comment by funder7** [comment]
Hi @alexziskind1 actually this version is compatible with ns6, I made some confusion with versions :-) 

I think that it's safer to merge this into a new branch, instead of master directly... Now I'm using it on Android, but I've got no chance to test in on iOS!

File: src/package.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "nativescript-oauth2",
-  "version": "2.5.0",
+  "version": "2.5.0-rc.0",
```
Comment (funder7) [inline src/package.json:3]: @nativescript/core and types are not released as stable yet, I would follow that naming to let the user know about upcoming updates, once stable packages are released

**Comment by bastienlemaitre** [comment]
https://github.com/NativeScript/nativescript-angular/issues/2222

**Comment by alexziskind1** [comment]
Thanks for the PR. I'm going to close this out as I've done the update separately on a live stream so more folks could benefit (i hope). But I appreciate your contribution

**Pull request closed without merging**